packages/admin/admin/src/hooks/useAsyncOptionsProps.ts

```diff
@@ -11,22 +11,25 @@ export interface AsyncOptionsProps<T> {
 export function useAsyncOptionsProps<T>(loadOptions: () => Promise<T[]>): AsyncOptionsProps<T> {
     const [open, setOpen] = React.useState(false);
     const [options, setOptions] = React.useState<T[]>([]);
-    const loading = open && options.length === 0;
+    const [loading, setLoading] = React.useState(false);
+    //const loading = open && options.length === 0;
```

Comment:
Checking for options.length here results in never reloading the options even if `loadOptions` changes. My fix isn't the ideal solution since it always reloads on open, but better than before.

Reply:
I solved it this way: https://github.com/vivid-planet/comet/pull/2031/files#diff-3fcf564cf314eb70bd6da272ac6f982cc91707cad756c4c5a5b9e9f1623f0ecf

Reply:
I don't like having to call refetch manually, and I'm not even sure why this works at all 😅 

Options should be automatically reloaded when the loadOptions function changes IMO

Reply:
Solved this by checking for `open` instead of `loading` in the `useEffect`. IMO this is sufficient for real world use cases where the response of the query is cached. See the provided screencast for an example.

Or should I added a more sophisticated solution to only reload if `loadOptions` changed and the select is open?

Reply:
> this is sufficient for real world use cases where the response of the query is cached

you mean because you clear options and immediately set the options from cached response? and it doesn't flicker?

Reply:
> you mean because you clear options and immediately set the options from cached response? and it doesn't flicker?

Yes. See the screencast.